doc/kconfig.md : Update documentation (#14154)
RIOT-OS/RIOT: doc/doxygen/src/kconfig/kconfig.md

editing the source files will have no effect. To go back to manual edition
a `make clean` has to be issued in the application directory.

> ## Appendix D: A few key aspects while exposing a macro to Kconfig

Reviewer: ```suggestion
## Appendix D: A few key aspects while exposing a macro to Kconfig {#kconfig-appendix-d}
```

> A macro that holds a 0 or 1 gets modelled as a bool. References to this macro

Reviewer: ```suggestion
A macro that holds a 0 or 1 is modelled in Kconfig as a `bool` symbol. References to this macro
```
